src/components/uri/Visualization.tsx

```diff
@@ -246,24 +254,31 @@ export class Visualization extends React.Component<IVisualizationProps, IVisuali
                 return this.props.fetchVisObject(visualizationUri);
             })
             .then((mdObject: VisualizationObject.IVisualizationObject) => {
-                const translatedPopSuffix = ' - previous year'; // TODO change hardcoded PoP suffix with new visObj
-
-                const visObjTotals = mdObject.content.buckets.totals;
-                const totals = !isEmpty(visObjTotals) ?
-                    visObjTotals.map(visObjTotal => visObjTotal.total) : [];
-
-                const { afm, resultSpec } = toAfmResultSpec(mdObject.content, translatedPopSuffix);
-                const dateFilter = getDateFilter(filters);
-                const attributeFilters = getAttributeFilters(filters);
-                const afmWithFilters = AfmUtils.appendFilters(afm, attributeFilters, dateFilter);
-                return this.adapter.createDataSource(afmWithFilters)
-                    .then((dataSource: IDataSource) => {
-                        return {
-                            type: mdObject.content.type,
-                            dataSource,
-                            resultSpec,
-                            totals
-                        };
+                const visualizationClass: string = get(mdObject, 'content.visualizationClass.uri', '');
```

Comment:
move it to `src/helpers/MdObjectHelper.ts` ? It'll provide ability for better unit test coverage

Reply:
moved and added UTs